extensions/ui/org.eclipse.smarthome.ui.basic/web-src/smarthome.js

```diff
@@ -1192,21 +1192,32 @@
 				item: _t.item,
 				value: _t.input.checked ? "ON" : "OFF"
 			}));
-			_t.suppressUpdate();
 		});
 
+		_t.hasValue = _t.parentNode.getAttribute("data-has-value") === "true";
+		_t.valueNode = _t.parentNode.parentNode.querySelector(o.formValue);
+
 		_t.setValuePrivate = function(value, itemState) {
 			var
 				val = itemState === "ON";
 
-			_t.input.checked = val;
-
-			if (val) {
+			if (_t.input.checked !== val) {
+				_t.input.checked = val;
+				if (val) {
 				_t.parentNode.MaterialSwitch.on();
```

Comment:
If you use Notepad++, save as UTF-8 (without BOM) instead of UTF-8. Also, you can stage files partially in git.